### ISSUE TITLE
User Story: Switching to current best available communication mode between Vehicle and Service 

### ISSUE THREAD

**Issue by alldne** [gcivil-nyu-org/spring2020-cs-gy-9223-backend#22]
**User story**
As pit crew, I want to configure how to communicate between the vehicle and the service so that I could choose the best & available network option under various environments

**Acceptance criteria**
- The way of communication between vehicle and mercury is configurable, not hard-coded
- Configurable options are 4G and/or radio at this point

**Definition of Done**
I know this is done when I can dynamically configure the way of communication between the vehicle and the service

**Comment by VentusXu09** [comment]
I get confused with #24. What's the difference between the two tickets?

**Comment by Ricardo-1894** [comment]
> I get confused with #24. What's the difference between the two tickets?

I think this is more about communication configuration (system can identify the best communication for pit crew to configure), while #24 is about whether system can internally provide the right data stream even if there are multiple data streams.

**Comment by VentusXu09** [comment]
> > I get confused with #24. What's the difference between the two tickets?
> 
> I think this is more about communication configuration (system can identify the best communication for pit crew to configure), while #24 is about whether system can internally provide the right data stream even if there are multiple data streams.

So in this issue, users choose which communication methods are valid, and in #24 the system will automatically use the best method between valid ones.
Is that correct?

**Comment by alldne** [comment]
> So in this issue, users choose which communication methods are valid,

Yes. Because every motorsports team might have different network setup (radio and/or 4G), we should provide a way for them to configure this.

>  and in #24 the system will automatically use the best method between valid ones.

Not exactly. _"Automatically"_ is true but _"use the best method"_ seems you're assuming the system choose a single best connection method at that time. I think It's more like the system should listen to all the connection and combine them to get a best up-to-date data at that time.
